nipoppy_cli/nipoppy/workflows/pipeline.py

```diff
@@ -270,20 +270,22 @@ def set_up_bids_db(
         """Set up the BIDS database."""
         dpath_bids_db: Path = Path(dpath_bids_db)
 
+        pybids_ignore_patterns = self.pybids_ignore_patterns.copy()
+
         if participant_id is not None:
             add_pybids_ignore_patterns(
-                current=self.pybids_ignore_patterns,
+                current=pybids_ignore_patterns,
                 new=f"^(?!/{BIDS_SUBJECT_PREFIX}({participant_id}))",
             )
         if session_id is not None:
             add_pybids_ignore_patterns(
-                current=self.pybids_ignore_patterns,
+                current=pybids_ignore_patterns,
                 new=f".*?/{BIDS_SESSION_PREFIX}(?!{session_id})",
```

Comment:
Just out of curiosity, how come the regex patterns for `participant` and `session` are different?

Reply:
Participant is at the beginning of the string. Also the prefixes are different